pkg/rtc/transport.go

```diff
@@ -306,6 +335,96 @@ func (t *PCTransport) createAndSendOffer(options *webrtc.OfferOptions) error {
 	return nil
 }
 
+func (t *PCTransport) preparePC(previousAnswer webrtc.SessionDescription) error {
+	// sticky data channel to first m-lines, if someday we don't send sdp without media streams to
+	// client's subscribe pc after joining, should change this step
+	parsed, err := previousAnswer.Unmarshal()
+	if err != nil {
+		return err
+	}
+	fp, fpHahs, err := extractFingerprint(parsed)
+	if err != nil {
+		return err
+	}
+
+	offer, err := t.pc.CreateOffer(nil)
+	if err != nil {
+		return err
+	}
+
+	t.pc.SetLocalDescription(offer)
+
+	pc2, err := webrtc.NewPeerConnection(webrtc.Configuration{})
+	if err != nil {
+		return err
+	}
+	defer pc2.Close()
+
+	pc2.SetRemoteDescription(offer)
```

Comment:
@cnderrauber I cannot keep up with this mastery :-) Can you please write explicit comments about what each section or each magic line of code is doing inline? For e.g. why do we need `pc2`? What is the magic line which sticks data channel as the first media section? What is the finger print magic? etc.

Reply:
Add some comment for these, hope it is useful for understand how it works